resolwe_bio/tools/goea.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+"""Map knowledge base features."""
+from subprocess import Popen, PIPE, DEVNULL
+
+import argparse
+import json
+import tempfile
+import resdk
+
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Map feature IDs between different databases.")
+    parser.add_argument('url', type=str, help="Knowledge base URL")
+    parser.add_argument('source_db', type=str, help="Source database.")
+    parser.add_argument('target_db', type=str, help="Target database.")
+    parser.add_argument('species', type=str, help="Species.")
+    parser.add_argument('feature_ids', help="Feature IDs file")
+    parser.add_argument('obo', help="Ontology object file")
+    parser.add_argument('gaf', help="GAF annotation file")
+    parser.add_argument('--pval', type=float, default=0.1, help="P-value threshold")
+    parser.add_argument('--min_genes', type=int, default=1, help="Minimum number of genes on a GO term.")
+    return parser.parse_args()
+
+
+def main():
+    """Invoked when run directly as a program."""
+    args = parse_arguments()
+
+    res = resdk.Resolwe(url=args.url)
+
+    with open(args.feature_ids) as gene_file:
+        genes = [gene for gene in gene_file]
+
+    org_features = res.feature.filter(source=args.source_db, query=genes)
+
+    if len(org_features) == 0:
+        print('{"proc.error":"No genes were fetched from the knowledge base."}')
+        exit(1)
+
+    species = list(set([feature.species for feature in org_features]))
+
+    if len(species) != 1:
+        print('{"proc.error":"Input genes belong to multiple species."}')
+        exit(1)
+    else:
+        species = species[0]
```

Comment:
Use `set(feature.species for feature in org_features)` in line 41 instead of `list(set([feature.species for feature in org_features]))`. Then use here `species  = species.pop()`